plugin-variables-helper.ts

```diff
@@ -0,0 +1,93 @@
+	 * @param {string} configuration The configuration for which the variable will be used.
+	 * @returns {any} The value of the plugin variable specified in --var or undefined.
+	 */
+	public getPluginVariableFromVarOption(variableName: string, configuration?: string): any {
+		let varOption = this.$options.var;
+		configuration = configuration ? configuration.toLowerCase() : undefined;
+		let lowerCasedVariableName = variableName.toLowerCase();
+		if(varOption) {
+			let configVariableValue: string;
+			let generalVariableValue: string;
+			 if(variableName.indexOf(".") !== -1) {
+				varOption = this.simplifyYargsObject(varOption, configuration);
+			 }
+			_.each(varOption, (propValue: any, propKey: string) => {
+				if(propKey.toLowerCase() === configuration) {
+					_.each(propValue, (configPropValue: string, configPropKey: string) => {
+						if(configPropKey.toLowerCase() === lowerCasedVariableName) {
+							configVariableValue = configPropValue;
+							return false;
+						}
+					});
+				} else if(propKey.toLowerCase() === lowerCasedVariableName) {
+					generalVariableValue = propValue;
+				}
+			});
+
+			let value = configVariableValue || generalVariableValue;
+			if(value) {
+				let obj = Object.create(null);
+				obj[variableName] = value.toString();
+				return obj;
+			}
+		}
+
+		return undefined;
+	}
+
+	/**
+	 * Converts complicated yargs object with many subobjects, to simplified one.
+	 * Use it when the plugin variable contains dots ("."). In this case yargs treats them as inner object instead of propery name.
+	 * For ex. '--var.debug.DATA.APP.ID testId' will be converted to {debug: {DATA: {APP: {ID: testId}}}}, while we need {debug: {DATA.APP.ID: testId}}
+	 * '--var.DATA.APP.ID testId' will be converted to DATA: {APP: {ID: testId}}}, while we need {DATA.APP.ID: testId}
+	 * @param {any} obj varObject created by yargs
+	 * @param {string} configuration The configuration for which the plugin variable will be used.
+	 * @return {any} Converted object if the obj paramater is of type object, otherwise - the object itself.
+	 */
+	public simplifyYargsObject(obj: any, configuration?: string): any {
+		if(obj && typeof(obj) === "object") {
+			let convertedObject:any = Object.create({});
+
+			_.each(obj, (propValue: any, propKey: string) => {
+				if(typeof(propValue) !== "object") {
+					convertedObject[propKey] = propValue;
+					return false;
+				}
+
+				configuration = configuration ? configuration.toLowerCase() : undefined;
+				let innerObj = this.simplifyYargsObject(propValue, configuration);
+
+				if(propKey.toLowerCase() === configuration) {
+					// for --var.debug.DATA.APP.ID testId
+					convertedObject[propKey] = innerObj;
+				} else {
+					// for --var.DATA.APP.ID testId
+					_.each(innerObj, (innerPropValue: any, innerPropKey: string) => {
+						convertedObject[`${propKey}.${innerPropKey}`] = innerPropValue;
+					});
+				}
+
+			});
+
+			return convertedObject;
+		}
+
+		return obj;
```

Comment:
If we call this function with a number, it will return the number unchanged. Is this the proper behavior? Shouldn't we assert on non-object arguments?